Core/Src/can_handler.c

```diff
@@ -1,11 +1,69 @@
-#include "analyzer.h"
-#include "ringbuffer.h"
 #include "can_handler.h"
-#include "shep_tasks.h"
 #include <stdio.h>
+#include <assert.h>
+#include <stdlib.h>
 
-ringbuffer_t *can1_rx_queue = NULL;
-ringbuffer_t *can2_rx_queue = NULL;
+#define CAN_MSG_QUEUE_SIZE 50 /* messages */
+
+#define CAN_DISPATCH_FLAG 1U
+
+#define NEW_CAN_MSG_FLAG 1U
+
+static osMessageQueueId_t can_outbound_queue;
+static osMessageQueueId_t can_inbound_queue;
+
+can_t *can1;
+can_t *can2;
+
+static uint32_t can1_id_list[] = {
+	//CANID_X,
+	0x002
+};
+
+static uint32_t can2_id_list[] = {
+	//CANID_X,
+	0x18FF50E5
+};
+
+osStatus_t queue_and_set_flag(osMessageQueueId_t queue, const void *msg_ptr,
+			      osThreadId_t thread_id, uint32_t flags)
+{
+	osStatus_t status = osMessageQueuePut(queue, msg_ptr, 0U, 0U);
```

Comment:
Since we use try semantics here, can we add some sort of error handling for when the queue is full? We can use a print for now.

Reply:
Oh ya good idea.  We could do that in cerb too.